src/main/java/org/example/dao/RouteDAOImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+import org.example.model.Route;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RouteDAOImpl implements RouteDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public RouteDAOImpl() {
+    }
+    @Override
+    public void insertRoute(Route route) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("INSERT INTO routes (startPointId, endPointId, shortestDistance) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
+            statement.setInt(1, route.getStartPoint().getId());
+            statement.setInt(2, route.getEndPoint().getId());
+            statement.setLong(3, route.getShortestDistance());
+            statement.executeUpdate();
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    int id = generatedKeys.getInt(1);
+                    route.setId(id);
+                    System.out.println("Route inserted successfully. Generated ID: " + id);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void updateRoute(Route route) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("UPDATE routes SET shortestDistance = ? WHERE startPointId = ? AND endPointId = ?")) {
+            statement.setLong(1, route.getShortestDistance());
+            statement.setInt(2, route.getStartPoint().getId());
+            statement.setInt(3, route.getEndPoint().getId());
+            int rowsAffected = statement.executeUpdate();
+            if (rowsAffected > 0) {
+                System.out.println("Route updated successfully.");
+            } else {
+                System.out.println("Route not found.");
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public Route getRoute(int id) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("SELECT startPointId as r.startPoint, endPointId as r.endPoint, shortestDistance as r.distance FROM routes r WHERE id = ?")) {
```

Comment:
This query doesn't work against our schema, the correct one could be
SELECT r.start_point AS startPointId, r.end_point AS endPointId, r.distance AS shortestDistance FROM routes r WHERE id = ?